### PR TITLE
VRF model object lists

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Use ModelObjectList for loading indexes instead of extensible list.

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp [hunk 1]
```diff
@@ -244,6 +211,12 @@ namespace model {
         result.push_back(curve.get());
       }
 
+      if (auto const _loadingIndexList = loadingIndexList()) {
+        for (const auto& mo : _loadingIndexList->modelObjects()) {
+          result.push_back(mo);
+        }
+      }
```
Comment (jmarrec) [inline src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp:218]: pedantic man to the rescue!

```suggestion
      if (auto const _loadingIndexList = loadingIndexList()) {
        auto mos =  _loadingIndexList->modelObjects();
        result.reserve(result.size() + mos.size());
        result.insert(result.end(), std::make_move_iterator(mos.begin()), std::make_move_iterator(mos.end()));
      }
```

Anyways, what makes me tick is the fact that `_loadingIndexList` is an optional, it shouldn't be. An please use a trailing optional_, not _optional, technically it's UB as anything starting with an underscore is reserved for the STL implementators. (I am probably the main person to blame for this pattern of a leading underscore for optional...)

File: src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp [hunk 2]
```diff
@@ -563,97 +536,69 @@ namespace model {
       return value.get();
     }
 
-    unsigned int AirConditionerVariableRefrigerantFlowFluidTemperatureControl_Impl::numberofCompressorLoadingIndexEntries() const {
-      return numExtensibleGroups();
+    boost::optional<ModelObjectList> AirConditionerVariableRefrigerantFlowFluidTemperatureControl_Impl::loadingIndexList() const {
+      return getObject<ModelObject>().getModelObjectTarget<ModelObjectList>(OS_AirConditioner_VariableRefrigerantFlow_FluidTemperatureControlFields::LoadingIndexList);
```
Comment (jmarrec) [inline src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp:540]:
```suggestion
    boost::optional<ModelObjectList> AirConditionerVariableRefrigerantFlowFluidTemperatureControl_Impl::loadingIndexList() const {
      boost::optional<ModelObjectList> mo = getObject<ModelObject>().getModelObjectTarget<ModelObjectList>(OS_AirConditioner_VariableRefrigerantFlow_FluidTemperatureControlFields::LoadingIndexList);
      OS_ASSERT(mo);
      return mo.get();
```

Comment (jmarrec) [reply src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp:540]: + add it in Ctor

File: src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp (hunk 1 above)
Comment (joseph-robertson) [reply src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp:218]: I took this straight from `CoilHeatingDXVariableSpeed.cpp`...

Comment (jmarrec) [reply src/model/AirConditionerVariableRefrigerantFlowFluidTemperatureControl.cpp:218]: hehe. Well the Ctor there does create the MoList... I think making non optional + OS_ASSERT when fetching it makes more sense. As far as make_move_iterator, it's just a side note, ignore it.

**Pull request merged**